specifyweb/frontend/js_src/lib/components/QueryBuilder/Formatter.tsx

```diff
@@ -173,13 +188,28 @@ function FormatSelect({
             className={customSelectElementBackground}
             disabled={handleChange === undefined}
             id={id('list')}
-            value={currentFormat}
-            onValueChange={handleChange}
+            value={compositeValue}
+            onValueChange={(value) => {
+              const parts = value.split('|||');
```

Comment:
In a similar vein, can you make `|||` a constant and reuse that here and in the function you create? It would help document the purpose of the string and keep any future changes more isolated.